cordova-common/spec/events.spec.js

```diff
@@ -0,0 +1,30 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var events = require('../src/events');
+
+describe('forwardEventsTo method', function () {
+    afterEach(function() {
+        events.forwardEventsTo(null);
+    });
+    it('should not go to infinite loop when trying to forward to self', function () {
+        events.forwardEventsTo(events);
+        events.emit('log', 'test message');
```

Comment:
This test will fail if we go into an infinite recursion as the title says. 
I will add another one for the `EVENTS_RECEIVER` internals check - although would it be better not to use rewire but to verify that forwarding has been reset after `events.forwardEventsTo(events)` call (the behavior change we are discussing below)?